pkg/tsdb/prometheus/prometheus.go

```diff
@@ -82,7 +101,7 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 			}
 		}
 
-		client, err := createClient(settings.URL, httpCliOpts, httpClientProvider)
+		client, err := createClient(settings.URL, httpCliOpts, httpClientProvider, forceHttpGet(jsonData))
```

Comment:
Nit. Would suggest unmarshaling json data into struct holding time interval and http method. 

If datasource settings is updated a new client will be created here so should work as expected.

Reply:
thanks for the tip... as this is going to get backported, i do not want to reach into those structures deeper, i will do that refactoring in a separate PR that will not get backported.